src/RayTracer/Public/js/render-3d.js

```diff
@@ -0,0 +1,161 @@
+
+		// when the mouse moves, call the given function
+		document.addEventListener('mousedown', onDocumentMouseDown, false);
+
+	}
+
+	function onDocumentMouseDown(event) {
+		// the following line would stop any other event handler from firing
+		// (such as the mouse's TrackballControls)
+		// event.preventDefault();
+
+
+		// update the mouse variable
+		mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
+		mouse.y = - (event.clientY / window.innerHeight) * 2 + 1;
+
+		// find intersections
+
+		// create a Ray with origin at the mouse position
+		//   and direction into the scene (camera direction)
+		var vector = new THREE.Vector3(mouse.x, mouse.y, 1);
+		projector.unprojectVector(vector, camera);
+		var ray = new THREE.Raycaster(camera.position, vector.sub(camera.position).normalize());
+
+		// create an array containing all objects in the scene with which the ray intersects
+		var intersects = ray.intersectObjects(targetList);
+
+		//if we want to use intersections in the future
+		/* // if there is one (or more) intersections
+		if (intersects.length > 0) {
+			console.log("Hit @ " + toString(intersects[0].point));
+			// change the color of the closest face.
+			intersects[0].face.color.setRGB(0.8 * Math.random() + 0.2, 0, 0);
+			intersects[0].object.geometry.colorsNeedUpdate = true;
+		} */
+
+	}
+
+	function toString(v) { return "[ " + v.x + ", " + v.y + ", " + v.z + " ]"; }
+
+
+	function create_cube(id, color, size, material, point) {
+		var idGeometry = new THREE.CubeGeometry(size.x, size.y, size.z);
+		let hex = rgbToHex(color.r, color.g, color.b);
+		var idMaterial = new THREE.MeshBasicMaterial({ color: hex, side: THREE.BackSide });
+		var id = new THREE.Mesh(idGeometry, idMaterial);
+		id.position.set(point.x, point.y, point.z);
+		scene.add(id);
+		targetList.push(id);
+	}
+
+	function create_sphere(id, color, size, material, point) {
+		let hex = rgbToHex(color.r, color.g, color.b);
+		var faceColorMaterial = new THREE.MeshBasicMaterial({ color: hex, vertexColors: THREE.FaceColors });
+		var sphereGeometry = new THREE.SphereGeometry(size.x, size.y, size.z);
+		var id = new THREE.Mesh(sphereGeometry, faceColorMaterial);
+		id.position.set(point.x, point.y, point.z);
+		scene.add(id);
+		targetList.push(id);
+	}
+
+	exports.jsonToShape = function (json) {
+		for (i = 0; i < json.length; i++) {
+
+			obj_color = json[i].color;
+			obj_material = json[i].material;
+			obj_point = json[i].point;
+			obj_size = json[i].size;
+			obj_id = json[i].id;
+
+			if (json[i].shape == "Sphere") {
+				create_sphere(obj_id, obj_color, obj_size, obj_material, obj_point);
+			}
+			else if (json[i].shape == "Cube") {
+				create_cube(obj_id, obj_color, obj_size, obj_material, obj_point);
+			}
+		}
+	}
+
+	exports.animate = function () {
+		requestAnimationFrame(exports.animate);
+		exports.render();
+		update();
+	}
+
+
+	function update() {
+		controls.update();
+	}
+
+	exports.render = function render() {
```

Comment:
If you want to make me super happy it would be amazing to refactor the file to have function signatures such as:
`function render(){
...
}`
and then at the bottom
`module.exports = {
   ...,
  update
}`

But this is also a valid way of doing it (and the way I showed you before) so no worries if you don't feel like changing it, both are good (sorry for changing it, I decided I like the 2nd way more but like I said, not a must for the review)